File: Baekjoon/10828.py
#link : https://www.acmicpc.net/problem/10828
import sys
input = sys.stdin.readline

l = int(input().strip())
arr = []

for iter in range(l):
    a = input().strip().split(' ')
    if(a[0] == 'push'):
        arr.append(a[1])
        
    elif(a[0] == 'pop' or a[0] == 'top'):
        if(len(arr) == 0) :
            print('-1')
            continue
        print(arr[-1])
        if(a[0] == 'pop') : del arr[-1]
        
    elif(a[0] == 'empty'):
        print('1' if len(arr) == 0 else '0')
        
    else:
        print(len(arr))

File: Baekjoon/10828.cpp
#include<iostream>
#include<cstring>
#include<stack>
using namespace std;

int main(void){
	int n;
	cin >> n;
	
	stack<int> s;
	char command[10];
	
	for(int i = 0 ; i < n ; i++){
		cin >> command;
		if(!strcmp(command, "push")){
			int a;
			cin >> a;
			s.push(a);
		}
		else if(!strcmp(command, "pop")){
			if(s.empty()) cout << "-1\n";
			else{
				int a = s.top();
				cout << a << "\n";
				s.pop();	
			}
		}
		else if(!strcmp(command, "size")) cout << s.size() << "\n";
		else if(!strcmp(command, "empty")) cout << s.empty() << "\n";
		else if(!strcmp(command, "top")){
			if(s.empty()) cout << "-1\n";
			else cout << s.top() << "\n";
		} 
	}
	
	return 0;
}
